core/src/wallet/bitcoin/database/BitcoinLikeTransactionDatabaseHelper.cpp

```diff
@@ -215,20 +221,25 @@ namespace ledger {
 
             // Fetch outputs
             rowset<soci::row> outputRows = (sql.prepare <<
-                    "SELECT idx, amount, script, address, transaction_uid FROM bitcoin_outputs WHERE transaction_hash = :hash "
+                    "SELECT idx, amount, script, address, transaction_uid, block_height FROM bitcoin_outputs WHERE transaction_hash = :hash "
                     "ORDER BY idx", use(out.hash)
             );
+
+            auto btcTxUid = BitcoinLikeTransactionDatabaseHelper::createBitcoinTransactionUid(accountUid, out.hash);
             for (auto& outputRow : outputRows) {
                 //Check if the output belongs to this account
                 //solve case of 2 accounts in DB one as sender and one as receiver
                 //of same transaction (filter on account_uid won't solve the issue because
                 //bitcoin_outputs going to external accounts have NULL account_uid)
-                if (outputRow.get<std::string>(4) == BitcoinLikeTransactionDatabaseHelper::createBitcoinTransactionUid(accountUid, out.hash)) {
+                if (outputRow.get<std::string>(4) == btcTxUid) {
                     BitcoinLikeBlockchainExplorerOutput output;
                     output.index = (uint64_t) outputRow.get<int>(0);
                     output.value.assignScalar(outputRow.get<long long>(1));
                     output.script = outputRow.get<std::string>(2);
                     output.address = outputRow.get<Option<std::string>>(3);
+                    if (outputRow.get_indicator(4) != i_null) {
```

Comment:
I’m not sure this is needed if you add a non-null column in SQL (that would require a reset of the database on the client, though). Up to you.

Reply:
Yeah that's why I didn't do so, I prefer not to force clients to reset their DB